### PR TITLE
Restore 'nucypher status'

### LINKED ISSUE #656: Fail gracefully after running `nucypher status` with a fresh install
Sentry Issue: [NUCYPHER-P6](https://sentry.io/nucypher/nucypher/issues/839463245/?referrer=github_integration)

```
FileNotFoundError: [Errno 2] No such file or directory: '/Users/david/Library/Application Support/nucypher/ursula.config'
(6 additional frame(s) were not displayed)
...
  File "click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "nucypher/cli/main.py", line 130, in status
    ursula_config = UrsulaConfiguration.from_configuration_file(filepath=config_file)
  File "nucypher/config/node.py", line 353, in from_configuration_file
    payload = cls._read_configuration_file(filepath=filepath)
  File "nucypher/config/node.py", line 335, in _read_configuration_file
    with open(filepath, 'r') as file:
```

### PR DESCRIPTION
**Pull request by KPrasch**
Fixes #656 https://github.com/nucypher/nucypher/issues/1136 - Based on #1064

### CODE REVIEW

File: nucypher/blockchain/eth/clients.py [hunk 1]
```diff
@@ -329,6 +342,27 @@ def sync(self, *args, **kwargs) -> bool:
...
+    is_local = False
+
+    def unlock_account(self, address, password):
+        return True
+
+    def sync(self, *args, **kwargs):
+        return True
+
+
+class InfuraClient(Web3Client):
```
Comment (derekpierre) [inline nucypher/blockchain/eth/clients.py]: Did a merge cause this class to be defined twice?

File: nucypher/blockchain/eth/clients.py [hunk 2]
```diff
@@ -377,8 +411,8 @@ class NuCypherGethProcess(LoggingMixin, BaseGethProcess):
     IPC_PROTOCOL = 'http'
     IPC_FILENAME = 'geth.ipc'
     VERBOSITY = 5
-    CHAIN_ID = NotImplemented  # 1
-    _CHAIN_NAME = 'mainnet'
+    CHAIN_ID = NotImplemented
+    _CHAIN_NAME = NotImplemented
```
Comment (derekpierre) [inline nucypher/blockchain/eth/clients.py]: 👍

File: nucypher/blockchain/eth/providers.py [hunk 3]
```diff
@@ -31,13 +32,23 @@ def _get_websocket_provider(provider_uri):
...
+        from web3.auto.infura.goerli import w3
+
+    except InfuraKeyNotFound:
+        raise ProviderError(f'{infura_envvar} must be provided in order to use an Infura Web3 provider.')
+
+    # Verify Connection
     connected = w3.isConnected()
     if not connected:
-        raise ProviderError('Cannot auto-detect node.  Provide a full URI instead.')
+        raise ProviderError('Failed to connect to Infura node.')
```
Comment (derekpierre) [inline nucypher/blockchain/eth/providers.py]: Would be nice to know what the provided uri was in the error.

```suggestion
        raise ProviderError(f'Failed to connect to Infura node {provider_uri}.')
```

File: nucypher/blockchain/eth/providers.py [hunk 4]
```diff
@@ -31,13 +32,23 @@ def _get_websocket_provider(provider_uri):
...
+    uri_breakdown = urlparse(provider_uri)
+    infura_envvar = 'WEB3_INFURA_PROJECT_ID'
+    os.environ[infura_envvar] = os.environ.get(infura_envvar, uri_breakdown.netloc)
+
+    try:
+        # TODO: Only testnet for now
+        from web3.auto.infura.goerli import w3
+
+    except InfuraKeyNotFound:
+        raise ProviderError(f'{infura_envvar} must be provided in order to use an Infura Web3 provider.')
```
Comment (derekpierre) [inline nucypher/blockchain/eth/providers.py]: same comment as below about uri.

```suggestion
        raise ProviderError(f'{infura_envvar} must be provided in order to use an Infura Web3 provider {provider_uri}.')
```

**Comment by cygnusv** [comment]
This is marked as draft. Is it ready for review?

**Comment by derekpierre** [comment]
Haha I guess I didn't pay attention to the status 😊, I  just the email about the request for review and proceeded.

**Comment by cygnusv** [comment]
I'm sure your review is well received, regardless of the status 😀 rather, I'm asking to know if I should be reviewing too.

File: nucypher/blockchain/eth/agents.py [hunk 5]
```diff
@@ -185,6 +187,14 @@ def get_current_period(self) -> int:
     # StakingEscrow Contract API
     #
 
+    def get_all_locked_tokens(self, periods: int = 1) -> int:
```
Comment (cygnusv) [inline nucypher/blockchain/eth/agents.py]: Why `periods=1` is the default and not 0, like `get_locked_tokens()`?

File: nucypher/cli/painting.py [hunk 6]
```diff
@@ -167,36 +168,35 @@ def paint_known_nodes(emitter, ursula) -> None:
...
 
-Provider URI ............. {provider_uri}
-Registry Path ............ {registry_filepath}
+    contract_payload = f"""
+| NuCypher Contracts |
 
-NucypherToken ............ {token}
-StakingEscrow ............ {escrow}
-PolicyManager ............ {manager}
+Chain .....................{blockchain.client.chain_name}
```
Comment (cygnusv) [inline nucypher/cli/painting.py]:
```suggestion
Chain .................... {blockchain.client.chain_name}
```

File: nucypher/cli/painting.py [hunk 7]
```diff
@@ -167,36 +168,35 @@ def paint_known_nodes(emitter, ursula) -> None:
...
-Gas Price ................ {gas_price}
-Active Staking Ursulas ... {ursulas}
+    network_payload = f"""
+| Staking |
 
-    """.format(period=ursula_config.staking_agent.get_current_period(),
-               gas_price=ursula_config.blockchain.client.gasPrice,
-               ursulas=ursula_config.staking_agent.get_staker_population())
+Current Period ........... {staking_agent.get_current_period()}
+Actively Staked Tokens.... {staking_agent.get_all_locked_tokens()}
```
Comment (cygnusv) [inline nucypher/cli/painting.py]: Print NU instead of Nunits.

File: nucypher/cli/painting.py [hunk 8]
```diff
@@ -167,36 +168,35 @@ def paint_known_nodes(emitter, ursula) -> None:
...
+Published Stakes ......... {staking_agent.get_staker_population()}
+Gas Price ................ {blockchain.client.gas_price}
```
Comment (cygnusv) [inline nucypher/cli/painting.py]: Print units of gas price. Perhaps GWei should be the default unit here?

File: nucypher/blockchain/eth/agents.py (hunk 5 above)
Comment (KPrasch) [reply nucypher/blockchain/eth/agents.py]: There is a slight misunderstanding in naming conventions here.  The signatures of `getAllLockedTokens` and `getLockedTokens` are similar but behave slightly differently internally.  See:  <https://github.com/nucypher/nucypher/blob/master/nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol#L313>

File: nucypher/blockchain/eth/clients.py (hunk 1 above)
Comment (KPrasch) [reply nucypher/blockchain/eth/clients.py]: :heavy_check_mark:

File: nucypher/blockchain/eth/providers.py (hunk 3 above)
Comment (KPrasch) [reply nucypher/blockchain/eth/providers.py]: :heavy_check_mark:

File: nucypher/blockchain/eth/providers.py (hunk 4 above)
Comment (KPrasch) [reply nucypher/blockchain/eth/providers.py]: :heavy_check_mark:

File: nucypher/cli/painting.py (hunk 6 above)
Comment (KPrasch) [reply nucypher/cli/painting.py]: :heavy_check_mark:

File: nucypher/cli/painting.py (hunk 7 above)
Comment (KPrasch) [reply nucypher/cli/painting.py]: :heavy_check_mark:

File: nucypher/cli/painting.py (hunk 8 above)
Comment (KPrasch) [reply nucypher/cli/painting.py]: :heavy_check_mark:

File: nucypher/blockchain/eth/agents.py (hunk 5 above)
Comment (cygnusv) [reply nucypher/blockchain/eth/agents.py]: Thanks for the clarification!

**Pull request merged**